hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2808,49 +2812,78 @@ public static HoodieData<HoodieRecord> convertMetadataToPartitionStatRecords(Hoo
       LOG.debug("Indexing following columns for partition stats index: {}", columnsToIndexSchemaMap.keySet());
       // Group by partitionPath and then gather write stats lists,
       // where each inner list contains HoodieWriteStat objects that have the same partitionPath.
-      List<List<HoodieWriteStat>> partitionedWriteStats = new ArrayList<>(allWriteStats.stream()
-          .collect(Collectors.groupingBy(HoodieWriteStat::getPartitionPath))
-          .values());
+      // TODO(yihua): only invoke FSV resolution if needed based on shouldScanColStatsForTightBound?
```

Comment:
`shouldScanColStatsForTightBound` needs to be cleaned up. 
it was lingering from previous attempt at partition stats. 
currently, everything is tight bound only. so, we can clean it up.

Reply:
Fixed.